### PR TITLE
Switch to Vanniktech for Maven publishing

### PR DESCRIPTION
**Pull request by swankjesse**
Steal configuration from Okio, which already works nicely.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -27,9 +29,10 @@ buildscript {
   }
 }
 
+apply(plugin = "com.vanniktech.maven.publish.base")
+
 allprojects {
   group = "com.squareup.okhttp3"
-  project.ext["artifactId"] = project.name.publishedArtifactId()
```
Comment (swankjesse) [inline build.gradle.kts:32]: I nuked this artifactId thing and went with the Gradle project name instead.

File: buildSrc/src/main/kotlin/deps.kt [hunk 2]
```diff
@@ -29,7 +29,7 @@ object Versions {
 
 object Dependencies {
   const val kotlinPlugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Versions.kotlin}"
-  const val dokkaPlugin = "org.jetbrains.dokka:dokka-gradle-plugin:0.10.1"
+  const val dokkaPlugin = "org.jetbrains.dokka:dokka-gradle-plugin:1.5.31"
```
Comment (swankjesse) [inline buildSrc/src/main/kotlin/deps.kt:32]: Big.

File: okcurl/build.gradle.kts [hunk 3]
```diff
@@ -66,3 +61,17 @@ graal {
...
+}
+
+tasks.register<Copy>("copyResourcesTemplates") {
+  from("src/main/resources-templates")
+  into("$buildDir/generated/resources-templates")
+  expand("projectVersion" to "${project.version}")
+  filteringCharset = StandardCharsets.UTF_8.toString()
+}.let {
+  tasks.processResources.dependsOn(it)
+  tasks.named("javaSourcesJar").dependsOn(it)
```
Comment (swankjesse) [inline okcurl/build.gradle.kts:76]: Without moving this task, javaSourcesJar doesn’t exist yet

File: settings.gradle.kts [hunk 4]
```diff
@@ -1,9 +1,13 @@
 rootProject.name = "okhttp-parent"
 
 include(":mockwebserver")
+project(":mockwebserver").name = "mockwebserver3"
```
Comment (swankjesse) [inline settings.gradle.kts:4]: Next step: rename the directories so I don’t need to rename the projects

**Comment by swankjesse** [comment]
In a follow-up: change the releasing process so builds happen on CI, not a developer’s computer.

File: mockwebserver-junit5/build.gradle.kts [hunk 5]
```diff
@@ -25,9 +30,7 @@ dependencies {
...
 }
 
-afterEvaluate {
-  tasks.dokka {
-    outputDirectory = "$rootDir/docs/4.x"
-    outputFormat = "gfm"
-  }
+configure<MavenPublishBaseExtension> {
+  configure(KotlinJvm(javadocJar = JavadocJar.Dokka("dokkaGfm"))
+  )
```
Comment (oldergod) [inline mockwebserver-junit5/build.gradle.kts:35]:
```suggestion
  configure(KotlinJvm(javadocJar = JavadocJar.Dokka("dokkaGfm")))
```

**Review by oldergod — APPROVED** [review]

**Review by yschimke — APPROVED** [review]

**Pull request merged**